src/software/world/field.h

```diff
@@ -35,6 +35,12 @@ typedef enum
 class Field
 {
    public:
+    /**
+     * Default constructs a field with all dimensions set to 0. This default
+     * field will be invalid (ie. isValid() returns false)
```

Comment:
What was the idea behind this default constructor?

Reply:
We have `isValid` again so have the ability to check for default-constructed / invalid fields. And we were _constantly_ constructing fields as `Field(0, 0, 0, 0, 0, 0, 0)` so I decided a default constructor was worth it again.